src/main/kotlin/com/ampnet/identityservice/service/impl/VerificationServiceImpl.kt

```diff
@@ -0,0 +1,65 @@
+package com.ampnet.identityservice.service.impl
+
+import com.ampnet.identityservice.exception.ErrorCode
+import com.ampnet.identityservice.exception.InvalidRequestException
+import com.ampnet.identityservice.exception.ResourceNotFoundException
+import com.ampnet.identityservice.service.VerificationService
+import mu.KLogging
+import org.kethereum.crypto.signedMessageToKey
+import org.kethereum.crypto.toAddress
+import org.kethereum.model.SignatureData
+import org.springframework.stereotype.Service
+import java.math.BigInteger
+import java.security.SecureRandom
+import java.security.SignatureException
+import kotlin.jvm.Throws
+
+@Service
+@Suppress("MagicNumber")
+class VerificationServiceImpl : VerificationService {
+
+    companion object : KLogging()
+
+    private val userPayload = mutableMapOf<String, String>()
+
+    override fun generatePayload(address: String): String {
+        // TODO generates a negative number, find a function in kethereum
+        val nonce = SecureRandom().nextLong().toString()
```

Comment:
Is this nonce important for security? If not, you can use ordinary `Random` which offers `nextLong(Long)` which returns number in range `[0, arg>` - `arg` is not included in the bound, but 0 is.

Reply:
It is important, it is returned to the frontend to sign it with its private key, after which the frontend requests jwt for this signature + his address.